packages/flutter_tools/lib/src/base/build.dart

```diff
@@ -308,7 +308,7 @@ class AOTSnapshotter {
       extraFrontEndOptions: extraFrontEndOptions,
       linkPlatformKernelIn: true,
       aot: true,
-      enableAsserts: buildMode == BuildMode.debug,
+      buildMode: buildMode,
       trackWidgetCreation: trackWidgetCreation,
       targetProductVm: buildMode == BuildMode.release,
```

Comment:
Maybe we can also get rid of 'targetProductVm' parameter.

Reply:
Done.